the-collab-lab/tcl-77-smart-shopping-list
# Issue#4: Set up new shopping list

## Description

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the Home view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the list path to localStorage
2. Save the list to our database using the createList function in firebase.js
3. Show the user the list view


## Related Issue


closes #4

## Acceptance Criteria


UI-related tasks:


Data-related tasks:


## Type of Changes

`new feature`

## Updates

### Before
<img width="605" alt="Screenshot 2024-08-22 at 9 46 26 PM" src="https://github.com/user-attachments/assets/5acb7caf-74d4-4678-a812-9490f6dfa586">




### After

https://github.com/user-attachments/assets/52c50113-795a-4cf8-a823-7e4640a84b1e



## Testing Steps / QA Criteria



1. Pull the branch locally using `git pull origin fz-ma-setup-new-shopping-list` in your terminal. 
2. Make sure you are signed into the app. 
3. On the Home Page, you will see a form with a field to enter the name of a new shopping list. Enter the name of your new list.
4. Clicking "Create List" will create the new list in the database, and will show an alert if the list was successfully created. 
5. The user is them redirected to the list view. 
6. Open up the console and check out local storage. The new list's path will show up there after successful creation.
